File: src/2525.categorize-box-according-to-criteria.2619/solution.ts
function categorizeBox (length: number, width: number, height: number, mass: number): string {
    const LIMIT4 = 10 ** 4;
    const LIMIT9 = 10 ** 9;

    const isBulky = (length >= LIMIT4 || width >= LIMIT4 || height >= LIMIT4 || mass >= LIMIT4) || length * height * width >= LIMIT9;
    const isHeavy = mass >= 100;
    if (isBulky && isHeavy) {
        return 'Both';
    }
    if (isBulky) {
        return 'Bulky';
    }
    if (isHeavy) {
        return 'Heavy';
    }
    return 'Neither';
}

File: src/2525.categorize-box-according-to-criteria.2619/solution.java
class Solution {
    private static int tenThousand = 10000;
    private static long billion = 1000000000;
    public String categorizeBox(int length, int width, int height, int mass) {
        boolean bulky = isBulky(length,width,height,mass);
        boolean heavy = mass>=100;

        if(bulky && heavy){
            return "Both";
        }
        if(bulky){
            return "Bulky";
        }
        if(heavy){
            return "Heavy";
        }
        return "Neither";
    }

    private boolean isBulky(int length, int width, int height, int mass){
        return (length>=tenThousand || width>=tenThousand || height>=tenThousand || mass>=tenThousand) || (long)length*height*width>=billion;
    }
}